retrievalmarket/impl/clientstates/client_fsm.go

```diff
@@ -214,10 +221,12 @@ var ClientEvents = fsm.Events{
 			rm.DealStatusFundsNeededLastPayment,
 			rm.DealStatusCheckComplete,
 			rm.DealStatusClientWaitingForLastBlocks).ToNoChange().
+		FromMany(rm.DealStatusSendFunds, rm.DealStatusSendFundsLastPayment).To(rm.DealStatusOngoing).
 		FromMany(paymentChannelCreationStates...).ToJustRecord().
 		Action(recordReceived),
 
 	fsm.Event(rm.ClientEventSendFunds).
+		FromMany(rm.DealStatusSendFunds, rm.DealStatusSendFundsLastPayment).To(rm.DealStatusOngoing).
```

Comment:
Why this particular transition ? So, we've processed a payment request and decided we need to send funds.

Now, if we were already in a state of sending funds, we are moving out of it. Why are we moving out of it ? 

1. What will go wrong if we don't move out of it ?
2. What about the fact that we do need to send money for this  `ClientEventSendFunds ` event ? Are you betting that going to the `Ongoing` state will again trigger a request to send money as I see that we do check for remaining payments after moving to the `Ongoing` state.